Assets/Scripts/ServiceManagers/ServiceManager.cs

```diff
@@ -0,0 +1,203 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+namespace PlayEveryWare.EpicOnlineServices
+{
+    using System;
+    using System.Collections.Generic;
+
+    using Epic.OnlineServices;
+    using Epic.OnlineServices.Connect;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// Contains implementation of common functionality between different
+    /// EOS Service managers (Currently EOSAchievementManager and StatsManager).
+    /// </summary>
+    public abstract class ServiceManager : IEOSOnConnectLogin, IEOSSubManager, IDisposable
+    {
+        /// <summary>
+        /// Stores a list of functions to be called whenever data related to
+        /// the manager has been updated.
+        /// </summary>
+        protected IList<Action> _updateCallbacks = new List<Action>();
+
+        /// <summary>
+        /// Indicates whether the service needs to have a user authenticated in
+        /// order to function properly.
+        /// </summary>
+        private bool _requiresAuthentication;
+
+        /// <summary>
+        /// Called when connect login has taken place.
+        /// </summary>
+        /// <param name="loginCallbackInfo"></param>
+        public void OnConnectLogin(LoginCallbackInfo loginCallbackInfo)
+        {
+            if (loginCallbackInfo.ResultCode != Result.Success)
+            {
+                UnityEngine.Debug.Log($"Player login did not succeed. Result code: {Enum.GetName(typeof(Result), loginCallbackInfo.ResultCode)}");
+                return;
+            }
+            OnPlayerLogin(loginCallbackInfo.LocalUserId);
+        }
+
+        /// <summary>
+        /// Base constructor for Service managers.
+        /// </summary>
+        /// <param name="requiresAuthentication">
+        /// Indicates whether the service manager requires a user to be
+        /// authenticated in order to function properly.
+        /// </param>
+        protected ServiceManager(bool requiresAuthentication)
+        {
+            _requiresAuthentication = requiresAuthentication;
+
+            EOSManager.Instance.AddConnectLoginListener(this);
+
+            _ = RefreshAsync();
+        }
+
+        /// <summary>
+        /// Determines whether a user is currently authenticated.
+        /// </summary>
+        /// <returns>True if a user is authenticated, false otherwise.</returns>
+        protected static bool IsAuthenticated()
+        {
+            return (IsAuthenticatedByEpicAuthInterface() || IsAuthenticatedByConnectInterface());
+        }
+
+        /// <summary>
+        /// Determines whether the user is currently authenticated using the
+        /// Auth Interface provided by the EOS SDK.
+        /// </summary>
+        /// <returns>
+        /// True if the user is authenticated using the Auth Interface, false
+        /// otherwise.
+        /// </returns>
+        private static bool IsLoggedInWithEpicAuthInterface()
```

Comment:
Because neither `EOSAchievementManager` nor `StatsManager` utilize logging in using the Auth Interface provided by Epic, this particular function should be removed for the time being. It's possible that in future iterations of this class it can be re-introduced, but adding it now lends confusion.

@andrew-hirata-playeveryware for visibility - since this is per discussion with them.

Reply:
Aforementioned changes have been made. Consequently, I am marking this conversation as 'resolved.'